config/config.go

```diff
@@ -217,9 +217,9 @@ func DecodeTypeAndName(key string) (typeStr configmodels.Type, fullName string,
 
 	// Create normalized fullName.
 	if nameSuffix == "" {
-		fullName = string(typeStr)
+		fullName = configmodels.FullName(typeStr)
 	} else {
-		fullName = string(typeStr) + typeAndNameSeparator + nameSuffix
+		fullName = configmodels.FullName(string(typeStr) + typeAndNameSeparator + nameSuffix)
```

Comment:
I would expose a ctor on the FullName that takes 2 params type and name. I would not spread the use of the separator, maybe parse should also be a method on full name.